canu/generate/switch/config/config.py

```diff
@@ -677,89 +747,60 @@ def generate_switch_config(
-            dell_config_hier = HConfig(host=dell_switch)
-            dell_config_hier.load_from_string(switch_config)
-            dell_config_hier.set_order_weight()
-            for line in dell_config_hier.all_children_sorted():
-                v1_config = v1_config + line.cisco_style_text() + "\n"
-        if "sw-spine" in node_shasta_name:
-            v1_config = ""
-            mellanox_switch = Host(node_shasta_name, "onyx", mellanox_options)
-            mellanox_config_hier = HConfig(host=mellanox_switch)
-            mellanox_config_hier.load_from_string(switch_config)
-            mellanox_config_hier.set_order_weight()
-            for line in mellanox_config_hier.all_children_sorted():
-                v1_config = v1_config + line.cisco_style_text() + "\n"
-        return v1_config, devices, unknown
-
-    if override:
+
+    def hier_options(switch_os):
         options_file = os.path.join(
             project_root,
             "canu",
             "validate",
             "switch",
             "config",
-            "options.yaml",
+            f"{switch_os}_options.yaml",
         )
-        if switch_name in override:
-            options = yaml.load(open(options_file))
-            host = Host(switch_name, "aoscx", options)
-            override_config = (
-                "# OVERRIDE CONFIG\n"
-                + "# The configuration below has been ignored and is not included in the GENERATED CONFIG\n"
-            )
-            override_config_hier = HConfig(host=host)
-            override_config_hier.load_from_string(switch_config)
-            override_config_hier.add_tags(override[switch_name])
-            # add ! to the end of the aruba banner.
-            banner = override_config_hier.get_child("contains", "banner")
-            banner.add_child("!")
-            for line in override_config_hier.all_children_sorted_by_tags(
-                "override",
-                None,
-            ):
-                override_config = override_config + "\n" + "#" + line.cisco_style_text()
-            override_config = override_config + "\n# GENERATED CONFIG\n"
-            for line in override_config_hier.all_children_sorted_by_tags(
-                None,
-                "override",
-            ):
-                # add two spaces to indented config to match aruba formatting.
-                if (
-                    line.cisco_style_text().startswith("  ")
-                    and "!" not in line.cisco_style_text()
-                ):
-                    override_config = (
-                        override_config + "\n" + "  " + line.cisco_style_text()
-                    )
-                else:
-                    override_config = (
-                        override_config + "\n" + line.cisco_style_text().lstrip()
-                    )
-
-            return override_config, devices, unknown
+        return options_file
+
+    if architecture == "network_v1":
```

Comment:
Not to be redone here, but as a general refactor we need to do, we have input like network_v1 or whatever from the command line, but we often also have more data from the model (when an shcd/ccj is presented).  Remember the model has that the node is a switch and all that.  I believe there is often enough metadata in the model that we should do a lookup there rather than if-thens w/ hard coded switch names and os weirdness.  

IMHO we should ONLY do hard coded name and other things when we are validating the SHCD or not using the shcd/ccj.  All other times we should be using the model for lookups.

It's a bad habit we've gotten into here that makes the code much harder to maintain over time.